crates/common/log_manager/src/log_utils.rs

```diff
@@ -153,6 +160,30 @@ pub fn filter_logs_path_on_metadata(
     }
 }
 
+fn temp_filename(logfile_path: &Path) -> Result<PathBuf, LogRetrievalError> {
+    let file_name = logfile_path
+        .file_name()
+        .ok_or_else(|| FileError::InvalidFileName {
+            path: logfile_path.to_path_buf(),
+            source: anyhow!("Does not name a valid file"),
```

Comment:
Why dou you need `anyhow!`?

Reply:
anyhow::Error is also like Box<dyn Error> + 'static, but also guarantees Send and Sync. I originally added it to InvalidFileName error variant because i wanted to use either io::Error or Utf8Error as a source, but I wouldn't say using `anyhow!` here to provide a a description of why is the filename invalid, to the user, is an antipattern. This could've just been a string, but because it's `anyhow::Error`, one can also use any other error as a source.
I addressed creating more enum variants in this comment: https://github.com/thin-edge/thin-edge.io/pull/2302#discussion_r1354377885.